examples/magento-graphcms/next.config.js

```diff
@@ -25,6 +42,58 @@ const nextConfig = {
   eslint: {
     ignoreDuringBuilds: true,
   },
+  // eslint-disable-next-line @typescript-eslint/require-await
+  headers: async () => [
+    {
+      headers: [
+        {
+          key: 'Cache-Control',
+          value: 'public, no-cache, no-store, max-age=0, must-revalidate',
+        },
+        {
+          key: 'CDN-Cache-Control',
+          value: 'public, max-age=31536000, stale-while-revalidate',
+        },
+      ],
+
+      source: '/:path*',
+    },
+    {
+      headers: [
+        {
+          key: 'Cache-Control',
+          value: 'public, max-age=31536000, immutable',
+        },
+        {
+          key: 'CDN-Cache-Control',
+          value: 'public, max-age=31536000, stale-while-revalidate',
+        },
+      ],
+
+      source:
+        '/:path(.+\\.(?:ico|png|svg|jpg|jpeg|gif|webp|json|js|css|mp3|mp4|ttf|ttc|otf|woff|woff2)$)',
+    },
+    {
+      headers: [
+        {
+          key: 'Cache-Control',
+          value: 'private, no-cache, no-store, max-age=0, must-revalidate',
+        },
+      ],
+
+      source: '/_next/data/:path*',
+    },
+    {
+      headers: [
+        {
+          key: 'Cache-Control',
+          value: 'private, no-cache, no-store, max-age=0, must-revalidate',
+        },
+      ],
+
+      source: '/_next/:path(.+\\.(?:json)$)',
+    },
+  ],
```

Comment:
What does this do, this seems a change of a _very_ large scope by modifying all headers for all requests?